### ISSUE TITLE
vcpkg build failed

### ISSUE THREAD

**Issue by UnaNancyOwen** [Microsoft/vcpkg#2427]
vcpkg (master/HEAD) build failed since db40d19.

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

~\Documents\GitHub> cd C:\
C:\> git clone https://github.com/Microsoft/vcpkg.git
Cloning into 'vcpkg'...
remote: Counting objects: 29853, done.
remote: Compressing objects: 100% (31/31), done.
remote: Total 29853 (delta 1), reused 1 (delta 0), pack-reused 29822
Receiving objects: 100% (29853/29853), 6.02 MiB | 2.90 MiB/s, done.
Resolving deltas: 100% (19743/19743), done.
C:\> cd vcpkg
C:\vcpkg [master ≡]> .\bootstrap-vcpkg.bat
Microsoft (R) Build Engine version 15.5.180.51428 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 12/23/2017 12:52:01 PM.
     1>Project "C:\vcpkg\toolsrc\dirs.proj" on node 1 (default targets).
     1>Project "C:\vcpkg\toolsrc\dirs.proj" (1) is building "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (2) on node 1 (Clean ta
       rget(s)).
     2>CoreClean:
         Creating directory "Release\".
     2>Project "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (2) is building "C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj" (3) on
       node 1 (Clean target(s)).
     3>CoreClean:
         Creating directory "Release\".
     3>Done Building Project "C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj" (Clean target(s)).
     2>Done Building Project "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (Clean target(s)).
     1>Project "C:\vcpkg\toolsrc\dirs.proj" (1) is building "C:\vcpkg\toolsrc\vcpkgmetricsuploader\vcpkgmetricsuploader.vc
       xproj" (4) on node 1 (Clean target(s)).
     4>CoreClean:
         Creating directory "Release\".
     4>Done Building Project "C:\vcpkg\toolsrc\vcpkgmetricsuploader\vcpkgmetricsuploader.vcxproj" (Clean target(s)).
     1>Project "C:\vcpkg\toolsrc\dirs.proj" (1) is building "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (2:2) on node 1 (Build
       target(s)).
   2:2>Project "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (2:2) is building "C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj" (3:4)
        on node 1 (default targets).
     3>PrepareForBuild:
         Creating directory "C:\vcpkg\toolsrc\Release\".
         Creating directory "Release\vcpkglib.tlog\".
       InitializeBuildStatus:
         Creating "Release\vcpkglib.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
       VcpkgTripletSelection:
         Using triplet "x86-windows" from "C:\vcpkg\installed\x86-windows\"
       ClCompile:
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\bin\HostX86\x86\CL.exe /
         c /I..\include /I"C:\vcpkg\installed\x86-windows\include" /Zi /nologo /W3 /WX- /diagnostics:classic /sdl /MP /O2
         /Oi /Oy- /GL /D DISABLE_METRICS=0 /D "VCPKG_VERSION=-2017-12-22-2d0a76370ead152896411d17aa19934235b93d1c" /D _MBC
         S /D NDEBUG /D _MBCS /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Yc"pch.h" /Fp"Releas
         e\vcpkglib.pch" /Fo"Release\\" /Fd"Release\vcpkglib.pdb" /Gd /TP /analyze- /errorReport:queue /std:c++latest ..\s
         rc\pch.cpp
         pch.cpp
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\bin\HostX86\x86\CL.exe /
         c /I..\include /I"C:\vcpkg\installed\x86-windows\include" /Zi /nologo /W3 /WX- /diagnostics:classic /sdl /MP /O2
         /Oi /Oy- /GL /D DISABLE_METRICS=0 /D "VCPKG_VERSION=-2017-12-22-2d0a76370ead152896411d17aa19934235b93d1c" /D _MBC
         S /D NDEBUG /D _MBCS /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Yu"pch.h" /Fp"Releas
         e\vcpkglib.pch" /Fo"Release\\" /Fd"Release\vcpkglib.pdb" /Gd /TP /analyze- /errorReport:queue /std:c++latest ..\s
         rc\vcpkg\base\checks.cpp ..\src\vcpkg\base\chrono.cpp ..\src\vcpkg\base\cofffilereader.cpp ..\src\vcpkg\base\enum
         s.cpp ..\src\vcpkg\base\files.cpp ..\src\vcpkg\base\lineinfo.cpp ..\src\vcpkg\base\machinetype.cpp ..\src\vcpkg\b
         ase\strings.cpp ..\src\vcpkg\base\system.cpp ..\src\vcpkg\binaryparagraph.cpp ..\src\vcpkg\build.cpp ..\src\vcpkg
         \commands.autocomplete.cpp ..\src\vcpkg\commands.buildexternal.cpp ..\src\vcpkg\commands.cache.cpp ..\src\vcpkg\c
         ommands.ci.cpp ..\src\vcpkg\commands.contact.cpp ..\src\vcpkg\commands.cpp ..\src\vcpkg\commands.create.cpp ..\sr
         c\vcpkg\commands.dependinfo.cpp ..\src\vcpkg\commands.edit.cpp ..\src\vcpkg\commands.env.cpp ..\src\vcpkg\command
         s.exportifw.cpp ..\src\vcpkg\commands.hash.cpp ..\src\vcpkg\commands.import.cpp ..\src\vcpkg\commands.integrate.c
         pp ..\src\vcpkg\commands.list.cpp ..\src\vcpkg\commands.owns.cpp ..\src\vcpkg\commands.portsdiff.cpp ..\src\vcpkg
         \commands.search.cpp ..\src\vcpkg\commands.upgrade.cpp ..\src\vcpkg\commands.version.cpp ..\src\vcpkg\dependencie
         s.cpp ..\src\vcpkg\export.cpp ..\src\vcpkg\globalstate.cpp ..\src\vcpkg\help.cpp ..\src\vcpkg\input.cpp ..\src\vc
         pkg\install.cpp ..\src\vcpkg\metrics.cpp ..\src\vcpkg\packagespec.cpp ..\src\vcpkg\packagespecparseresult.cpp ..\
         src\vcpkg\paragraphparseresult.cpp ..\src\vcpkg\paragraphs.cpp ..\src\vcpkg\parse.cpp ..\src\vcpkg\postbuildlint.
         buildtype.cpp ..\src\vcpkg\postbuildlint.cpp ..\src\vcpkg\remove.cpp ..\src\vcpkg\sourceparagraph.cpp ..\src\vcpk
         g\statusparagraph.cpp ..\src\vcpkg\statusparagraphs.cpp ..\src\vcpkg\triplet.cpp ..\src\vcpkg\update.cpp ..\src\v
         cpkg\userconfig.cpp ..\src\vcpkg\vcpkgcmdarguments.cpp ..\src\vcpkg\vcpkglib.cpp ..\src\vcpkg\vcpkgpaths.cpp ..\s
         rc\vcpkg\versiont.cpp
         checks.cpp
         chrono.cpp
         cofffilereader.cpp
         enums.cpp
         files.cpp
         lineinfo.cpp
         machinetype.cpp
         strings.cpp
         system.cpp
         binaryparagraph.cpp
         build.cpp
         commands.autocomplete.cpp
         commands.buildexternal.cpp
         commands.cache.cpp
         commands.ci.cpp
         commands.contact.cpp
         commands.cpp
         commands.create.cpp
         commands.dependinfo.cpp
         commands.edit.cpp
         commands.env.cpp
         commands.exportifw.cpp
         commands.hash.cpp
         commands.import.cpp
         commands.integrate.cpp
         commands.list.cpp
         commands.owns.cpp
         commands.portsdiff.cpp
         commands.search.cpp
         commands.upgrade.cpp
         commands.version.cpp
         dependencies.cpp
         export.cpp
         globalstate.cpp
     3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xmemory0(945): err
       or C2664: 'std::pair<const _Kty,_Ty>::pair(std::pair<const _Kty,_Ty> &&)': cannot convert argument 2 from 'std::str
       ing' to 'const vcpkg::VersionT &' [C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj]
                 with
                 [
                     _Kty=std::string,
                     _Ty=vcpkg::VersionT
                 ] (compiling source file ..\src\vcpkg\commands.portsdiff.cpp)
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xmemory0(943): n
         ote: Reason: cannot convert from 'std::string' to 'const vcpkg::VersionT' (compiling source file ..\src\vcpkg\com
         mands.portsdiff.cpp)
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xmemory0(943): n
         ote: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be ca
         lled (compiling source file ..\src\vcpkg\commands.portsdiff.cpp)
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xtree(766): note
         : see reference to function template instantiation 'void std::_Default_allocator_traits<_Alloc>::construct<_Ty,st
         d::string&,std::string&>(_Alloc &,_Objty *const ,std::string &,std::string &)' being compiled
                 with
                 [
                     _Alloc=std::allocator<std::_Tree_node<std::pair<const std::string,vcpkg::VersionT>,std::_Default_allo
         cator_traits<std::allocator<std::pair<const std::string,vcpkg::VersionT>>>::void_pointer>>,
                     _Ty=std::pair<const std::string,vcpkg::VersionT>,
                     _Objty=std::pair<const std::string,vcpkg::VersionT>
                 ] (compiling source file ..\src\vcpkg\commands.portsdiff.cpp)
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xtree(773): note
         : see reference to function template instantiation 'void std::_Default_allocator_traits<_Alloc>::construct<_Ty,st
         d::string&,std::string&>(_Alloc &,_Objty *const ,std::string &,std::string &)' being compiled
                 with
                 [
                     _Alloc=std::allocator<std::_Tree_node<std::pair<const std::string,vcpkg::VersionT>,std::_Default_allo
         cator_traits<std::allocator<std::pair<const std::string,vcpkg::VersionT>>>::void_pointer>>,
                     _Ty=std::pair<const std::string,vcpkg::VersionT>,
                     _Objty=std::pair<const std::string,vcpkg::VersionT>
                 ] (compiling source file ..\src\vcpkg\commands.portsdiff.cpp)
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xtree(961): note
         : see reference to function template instantiation 'std::_Tree_node<std::pair<const _Kty,_Ty>,std::_Default_alloc
         ator_traits<_Alloc>::void_pointer> *std::_Tree_comp_alloc<_Traits>::_Buynode<std::string&,std::string&>(std::stri
         ng &,std::string &)' being compiled
                 with
                 [
                     _Kty=std::string,
                     _Ty=vcpkg::VersionT,
                     _Alloc=std::allocator<std::pair<const std::string,vcpkg::VersionT>>,
                     _Traits=std::_Tmap_traits<std::string,vcpkg::VersionT,std::less<std::string>,std::allocator<std::pair
         <const std::string,vcpkg::VersionT>>,false>
                 ] (compiling source file ..\src\vcpkg\commands.portsdiff.cpp)
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xtree(962): note
         : see reference to function template instantiation 'std::_Tree_node<std::pair<const _Kty,_Ty>,std::_Default_alloc
         ator_traits<_Alloc>::void_pointer> *std::_Tree_comp_alloc<_Traits>::_Buynode<std::string&,std::string&>(std::stri
         ng &,std::string &)' being compiled
                 with
                 [
                     _Kty=std::string,
                     _Ty=vcpkg::VersionT,
                     _Alloc=std::allocator<std::pair<const std::string,vcpkg::VersionT>>,
                     _Traits=std::_Tmap_traits<std::string,vcpkg::VersionT,std::less<std::string>,std::allocator<std::pair
         <const std::string,vcpkg::VersionT>>,false>
                 ] (compiling source file ..\src\vcpkg\commands.portsdiff.cpp)
         ..\src\vcpkg\commands.portsdiff.cpp(105): note: see reference to function template instantiation 'std::pair<std::
         _Tree_iterator<std::_Tree_val<std::_Tree_simple_types<_Ty>>>,bool> std::_Tree<std::_Tmap_traits<_Kty,vcpkg::Versi
         onT,_Pr,_Alloc,false>>::emplace<std::string&,std::string&>(std::string &,std::string &)' being compiled
                 with
                 [
                     _Ty=std::pair<const std::string,vcpkg::VersionT>,
                     _Kty=std::string,
                     _Pr=std::less<std::string>,
                     _Alloc=std::allocator<std::pair<const std::string,vcpkg::VersionT>>
                 ]
         ..\src\vcpkg\commands.portsdiff.cpp(105): note: see reference to function template instantiation 'std::pair<std::
         _Tree_iterator<std::_Tree_val<std::_Tree_simple_types<_Ty>>>,bool> std::_Tree<std::_Tmap_traits<_Kty,vcpkg::Versi
         onT,_Pr,_Alloc,false>>::emplace<std::string&,std::string&>(std::string &,std::string &)' being compiled
                 with
                 [
                     _Ty=std::pair<const std::string,vcpkg::VersionT>,
                     _Kty=std::string,
                     _Pr=std::less<std::string>,
                     _Alloc=std::allocator<std::pair<const std::string,vcpkg::VersionT>>
                 ]
         help.cpp
         input.cpp
         install.cpp
         metrics.cpp
         packagespec.cpp
         packagespecparseresult.cpp
         paragraphparseresult.cpp
         paragraphs.cpp
         parse.cpp
         postbuildlint.buildtype.cpp
         postbuildlint.cpp
         remove.cpp
         sourceparagraph.cpp
         statusparagraph.cpp
         statusparagraphs.cpp
         triplet.cpp
         update.cpp
         userconfig.cpp
         vcpkgcmdarguments.cpp
         vcpkglib.cpp
         vcpkgpaths.cpp
         versiont.cpp
     3>Done Building Project "C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj" (default targets) -- FAILED.
     2>Done Building Project "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (Build target(s)) -- FAILED.
     1>Done Building Project "C:\vcpkg\toolsrc\dirs.proj" (default targets) -- FAILED.

Build FAILED.

       "C:\vcpkg\toolsrc\dirs.proj" (default target) (1) ->
       "C:\vcpkg\toolsrc\vcpkg\vcpkg.vcxproj" (Build target) (2:2) ->
       "C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj" (default target) (3:4) ->
       (ClCompile target) ->
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.12.25827\include\xmemory0(945): e
       rror C2664: 'std::pair<const _Kty,_Ty>::pair(std::pair<const _Kty,_Ty> &&)': cannot convert argument 2 from 'std::s
       tring' to 'const vcpkg::VersionT &' [C:\vcpkg\toolsrc\vcpkglib\vcpkglib.vcxproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:07.20
C:\vcpkg\scripts\bootstrap.ps1 : Building vcpkg.exe failed. Please ensure you have installed Visual Studio with the
Desktop C++ workload and the Windows SDK for Desktop C++.
At line:1 char:4
+ & {& 'C:\vcpkg\scripts\bootstrap.ps1'}
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,bootstrap.ps1

C:\vcpkg [master ≡]>
```
https://ci.appveyor.com/project/PointCloudLibrary/pcl/build/1.0.278

**Comment by UnaNancyOwen** [comment]
@ras0219-msft @alexkaratarakis Can you fix it?

**Comment by jasjuang** [comment]
I can confirm that bootstrap is failing for me as well

**Comment by atkawa7** [comment]
Confirmed as  well

**Comment by atkawa7** [comment]
@UnaNancyOwen @jasjuang @alexkaratarakis @ras0219-msft See #2429

**Comment by giorgio-denunzio** [comment]
Bootstrap failing for me too on one out of three computers, everything ok on the other two, all of them Windows10. The same C2664 message. 
Thanks. 
Giorgio

**Comment by pravic** [comment]
> 2d0a7637 2017-12-22 | clang-tidy fixes (origin/master)

Don't push commits on Friday's evening ;)

**Comment by ras0219-msft** [comment]
@pravic Yep, agreed.

I have reverted the offending commit while we investigate how to fix it properly (might be #2429, but reverting is safer to do immediately 😄).

Thank you everyone for posting about this and you have our sincerest apologies for any inconvenience!

**Comment by scanban** [comment]
@ras0219-msft , it's definitely near #2429 
corrected it: [https://github.com/scanban/vcpkg/commit/723f1dcd71a312cf7158e7268ffe7798d582fd00]()
